core/src/main/java/hudson/Launcher.java

```diff
@@ -142,6 +143,20 @@ public Computer getComputer() {
     }
 
     /**
+     * Gets the IP address of the host running the jenkins slave agent which do control this Launcher
+     * <p>
+     * In most cases, slave agent do run on localhost, but within a containerized architecture, the
+     * slave agent hostname may be a distinct one. A {@link LauncherDecorator} responsible to setup
+     * such an architecture could then decorate this method to return the adequate slave agent host address.
+     * <p>
+     * Default implementation do return wildcard address (0.0.0.0 or ::0)
+     */
+    public String getSlaveAgentHostAddress() throws IOException {
+        return new InetSocketAddress(0).getAddress().getHostAddress();
```

Comment:
Surely this should be wrapped in a `Callable` so that it runs on the slave rather than on master? (It seems to produce `0.0.0.0` when I try it on my Linux machine, but if that is guaranteed then we should just hardcode that string to begin with.)